uint16 data
float32 duration